From: Andrew Morton <akpm@linux-foundation.org>
Subject: memblock-unpreserve-memory-in-case-of-error-fix
Date: Mon Nov  3 07:23:35 PM PST 2025

s/err_no_fdt_page/err_report/ in prepare_kho_fdt(), per Mike

Cc: Mike Rapoport (Microsoft) <rppt@kernel.org>
Cc: Pasha Tatashin <pasha.tatashin@soleen.com>
Cc: Pratyush Yadav <pratyush@kernel.org>
Cc: Alexander Graf <graf@amazon.com>
Cc: Changyuan Lyu <changyuanl@google.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: Jason Gunthorpe <jgg@nvidia.com>
Cc: Jason Gunthorpe <jgg@ziepe.ca>
Cc: Jonathan Corbet <corbet@lwn.net>
Cc: Masahiro Yamada <masahiroy@kernel.org>
Cc: Miguel Ojeda <ojeda@kernel.org>
Cc: Randy Dunlap <rdunlap@infradead.org>
Cc: Simon Horman <horms@kernel.org>
Cc: Tejun Heo <tj@kernel.org>
Cc: Zhu Yanjun <yanjun.zhu@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memblock.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/memblock.c~memblock-unpreserve-memory-in-case-of-error-fix
+++ a/mm/memblock.c
@@ -2483,7 +2483,7 @@ static int __init prepare_kho_fdt(void)
 	fdt_page = alloc_page(GFP_KERNEL);
 	if (!fdt_page) {
 		err = -ENOMEM;
-		goto err_no_fdt_page;
+		goto err_report;
 	}
 
 	fdt = page_to_virt(fdt_page);
@@ -2527,7 +2527,7 @@ err_unpreserve_fdt:
 	kho_unpreserve_pages(fdt_page, 1);
 err_free_fdt:
 	put_page(fdt_page);
-err_no_fdt_page:
+err_report:
 	pr_err("failed to prepare memblock FDT for KHO: %d\n", err);
 
 	return err;
_
